bincrafters_envy/azure.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+
+from .base import Base
+from requests.auth import HTTPBasicAuth
+import json
+
+
+class Azure(Base):
+    name = "azure"
+    default_host = "https://dev.azure.com"
+
+    def __init__(self, config, host):
+        super(Azure, self).__init__(config, host)
+
+        self._account = self._read_account(config) or 'bincrafters'
+        self._endpoint = self._host + '/' + self._account + "/packages/_apis"
+        self._api_version = "?api-version=5.0"
+        self._headers = {'Content-type': 'application/json'}
+        self._auth = HTTPBasicAuth(username=self._token, password="")
+
+    def _get_default_branch(self, url):
+        import subprocess
+        import tempfile
+
+        # is there a better way, e.g. without clone?
```

Comment:
If the project is available on Github, you use their REST API to collect this information. Also, Conan tools provides a smooth helper to Git commands. Anyway, your solution works fine.

Reply:
yeah, I anyway will need to obtain GitHub token from the AZP in order to establish service connection